server/py/framework/db/sqldb/db.py

```diff
@@ -1781,6 +1840,19 @@ def _find_artifacts(
                 partition_order,
                 with_tagged=True,
             )
+        if parent_key or parent_tag:
+            ref_alias = aliased(ArtifactV2)
+
+            # Join on reference_artifact_id -> ArtifactV2.id
+            query = query.outerjoin(ref_alias, ArtifactV2.parent_id == ref_alias.id)
```

Comment:
I suggest to revisit the outerjoin here as it will multiple the artifacts table twice. perhaps you can use specific columns on retrieval so outerjoin wont create 2 full artifact tables

Reply:
Good catch... for sure it suppose to be inner join